lib/core/include/irods/irods_load_plugin.hpp

```diff
@@ -94,6 +101,274 @@ namespace irods
+                    lib.read(reinterpret_cast<char*>(&s_hdr_in), sizeof(s_hdr_in));
+
+                    sh_type = s_hdr_in.sh_type;
+                    if (sh_type != SHT_DYNAMIC) {
+                        continue;
+                    }
+                    sh_name = s_hdr_in.sh_name;
+                    s_name = std::string_view(&e_strtbl_vec.data()[sh_name]);
+                    if (!s_name.starts_with(".dynamic")) {
+                        continue;
+                    }
+
+                    sh_offset = s_hdr_in.sh_offset;
+                    sh_size = s_hdr_in.sh_size;
+                    sh_entsize = s_hdr_in.sh_entsize;
+                }
+                else {
+                    Elf64_Shdr s_hdr_in{};
+                    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+                    lib.read(reinterpret_cast<char*>(&s_hdr_in), sizeof(s_hdr_in));
+
+                    sh_type = s_hdr_in.sh_type;
+                    if (sh_type != SHT_DYNAMIC) {
+                        continue;
+                    }
+                    sh_name = s_hdr_in.sh_name;
+                    s_name = std::string_view(&e_strtbl_vec.data()[sh_name]);
+                    if (!s_name.starts_with(".dynamic")) {
+                        continue;
+                    }
+
+                    sh_offset = s_hdr_in.sh_offset;
+                    sh_size = s_hdr_in.sh_size;
+                    sh_entsize = s_hdr_in.sh_entsize;
+                }
+
+                const std::uint64_t sh_offset_end = sh_offset + sh_size;
+                for (std::uint64_t de_off = sh_offset; de_off < sh_offset_end; de_off += sh_entsize) {
+                    // seek to dynamic entry
+                    lib.seekg(de_off);
+
+                    std::uint64_t dt_flags_1{};
+                    if (ei_class == ELFCLASS32) {
+                        Elf32_Dyn de_in{};
+                        // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+                        lib.read(reinterpret_cast<char*>(&de_in), sizeof(de_in));
+
+                        if (de_in.d_tag != DT_FLAGS_1) {
+                            continue;
+                        }
+
+                        // NOLINTNEXTLINE(cppcoreguidelines-pro-type-union-access)
+                        dt_flags_1 = de_in.d_un.d_val;
+                    }
+                    else {
+                        Elf64_Dyn de_in{};
+                        // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+                        lib.read(reinterpret_cast<char*>(&de_in), sizeof(de_in));
+
+                        if (de_in.d_tag != DT_FLAGS_1) {
+                            continue;
+                        }
+
+                        // NOLINTNEXTLINE(cppcoreguidelines-pro-type-union-access)
+                        dt_flags_1 = de_in.d_un.d_val;
+                    }
+
+                    if ((dt_flags_1 & static_cast<std::uint64_t>(DF_1_NOW)) != 0U) {
```

Comment:
Just curious. Why `!= 0U` instead of `== DF_1_NOW`?

Are you interested in a subset of the bits being set?

Reply:
Yes, if any other flags are also set, `== DF_1_NOW` would return false, even if `DF_1_NOW` is set.

Reply:
Unless I'm misunderstanding, the bitwise-and with `DF_1_NOW` will only return the bits matching `DF_1_NOW`.

Reply:
oh, I misunderstood your question.

clang-tidy suggested `!= 0U`, so that's what I did. But `== DF_1_NOW` seems more correct. I'll make the changes.